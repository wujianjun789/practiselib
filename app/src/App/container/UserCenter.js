/**
 * create by Azrael on 2017/7/4
 */
import React, { Component } from 'react';
export default class UserCenter extends Component{
    constructor(props) {
        super(props);
        this.state = {
            active: false,
            list: [
                {name: '修改密码', key: 'alter'},
                {name: '退出管理系统', key: 'exit'}
            ]
        }
        this.userListToggle = this.userListToggle.bind(this);
    }

    userListToggle() {
        this.setState((prevState)=> {
            return {active: !prevState.active};
        });
    }

    render() {
        const {active, list} = this.state;
        const {className=""} = this.props;
        return (
            <div className={`header-right ${className}`} onClick={this.userListToggle}>
                <div className="user-icon clearfix"><span className="icon icon-user"></span></div>
                <ul className={`user-list ${active ? '' : 'hidden'}`}>
                {
                    list.map(item => <li key={item.key}><span className={`icon icon-${item.key}`}></span>{item.name}</li>)
                }
                </ul>
            </div>
        )
    }
}