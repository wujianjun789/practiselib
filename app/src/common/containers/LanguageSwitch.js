/**
 * Created by a on 2017/9/5.
 */
import React,{Component} from 'react';
import {getDefaultIntl} from '../../intl/index'
import {getObjectByKey} from '../../util/index'
export default class LanguageSwitch extends Component{
    constructor(props){
        super(props);
        this.state = {
            curLan:{id:"zh",name:"简体中文", path:"icon_chinese"},
            list:[
                {id:"zh", name:"简体中文", path:"icon_chinese"},
                {id:"en", name:"ENGLISH", path:"icon_english"}
            ]
        }

        this.onClick = this.onClick.bind(this);
    }

    componentWillMount(){
        getDefaultIntl((intl)=>{
            this.setState({curLan:getObjectByKey(this.state.list, 'id', intl.locale)});
        })
    }

    onClick(data){
        this.setState({curLan:data},()=>{
            window.setLanguage(data.id);
        });
    }

    render(){
        const {className=''} = this.props;
        const {curLan, list} = this.state;
        return <div className={"language-switch "+className}>
                <div className="lanauage-icon">
                    <span className={curLan.path}></span>
                </div>
                <ul className="list-group">
                    {
                        list.map(item=>{
                            return <li key={item.id} className="list-group-item" onClick={()=>this.onClick(item)}>
                                <span className={item.path}></span>
                                <span>{item.name}</span>
                            </li>
                        })
                    }
                </ul>
            </div>
    }
}