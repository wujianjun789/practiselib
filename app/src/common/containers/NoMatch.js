import React, { Component } from 'react';
export default class NoMatch extends Component {
    render() {
        return (
            <h1>
                404!
            </h1>
        )
    }
}