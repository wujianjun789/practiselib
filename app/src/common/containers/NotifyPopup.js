/**
 * Created by a on 2017/9/4.
 */
import React,{Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import '../../../public/styles/notify.less';

import {removeNotify} from '../actions/notifyPopup'
import {getNotifyStateClass} from '../../util/index'
export class NotifyPopup extends Component{
    constructor(props){
        super(props);
    }

    onClick(id){
        this.props.actions.removeNotify(id);
    }

    render(){
        const {notifyPopup} = this.props;

        let notifyList = notifyPopup.get("notifyList");
        let curList = notifyList.slice(0, 3);
        return <ul className="list-group notify-popup">
            {
                curList.map((item, index)=>{
                    let id = item.get("id");
                    let ani = item.get("animation")
                    return <li key={id} className={"list-group-item "+("notify-"+index)+" "+getNotifyStateClass(item.get("notifyType"))+" "+(ani ? "active":"")}>
                        {item.get("text")}
                        <span onClick={()=>this.onClick(id)}>&times;</span>
                    </li>
                })
            }
        </ul>
    }
}

function mapStateToProps(state) {
    return {
        notifyPopup: state.notifyPopup
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            removeNotify: removeNotify
        }, dispatch)
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(NotifyPopup);