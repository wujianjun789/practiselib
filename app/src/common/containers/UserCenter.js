/**
 * create by Azrael on 2017/7/4
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {overlayerShow, overlayerHide} from '../actions/overlayer';
import ConfirmPopup from '../../components/ConfirmPopup';
import AlterPwPopup from '../../components/AlterPwPopup';
import {confirmExit} from '../actions/userCenter';
import {modifyPassword} from '../../api/modifyPassword';
/**
 * @param {String} className  optional
 * @param {Object} router     isRequired
 */
export class UserCenter extends Component{
    constructor(props) {
        super(props);
        this.state = {
            active: false,
            list: [
                {name: '修改密码', key: 'alter',path:'#alter'},
                {name: '退出管理系统', key: 'exit',path:'#exit'}
            ]
        }
        this.userListToggle = this.userListToggle.bind(this);
        this.itemClick = this.itemClick.bind(this);
        this.cancel = this.cancel.bind(this);
        this.confirm = this.confirm.bind(this);
        this.AlterPwPopupConfirm = this.AlterPwPopupConfirm.bind(this);
    }

    userListToggle() {
        this.setState((prevState)=> {
            return {active: !prevState.active};
        });
        // if(this.state.active==false){
        //     setTimeout(() => {
        //         if(this.state.active==false){}else{
        //             this.setState((prevState)=> {
        //             return{active: !prevState.active}
        //             });
        //         }
        //     }, 3000);
        // }
    }

    cancel() {
        this.props.actions.overlayerHide();
    }

    confirm() {
        this.props.actions.confirmExit(() => {
            this.props.actions.overlayerHide();
        }, () => {
            this.props.actions.overlayerHide();
        });
    }

    AlterPwPopupConfirm(data) {
        modifyPassword(data,() => {
            this.props.actions.overlayerHide();
            this.props.router.push('/login');
        }, (err) => {
            console.log('密码错误');
        });
    }
 
    componentDidMount() {
    }

    itemClick(key) {
        let {overlayerHide, overlayerShow} = this.props.actions;
        if(key == 'alter') {
            overlayerShow(<AlterPwPopup className='alter-pw-popup' overlayerShow={overlayerShow} overlayerHide={overlayerHide} onConfirm={this.AlterPwPopupConfirm}/>);
        } else {
            overlayerShow(<ConfirmPopup tips="是否退出？" iconClass="icon-popup-exit" cancel={this.cancel} confirm={this.confirm}/>);
        }
    }

    render() {
        const {active, list} = this.state;
        const {className=""} = this.props;
        return (
            <div className={`user-center ${className}`} onClick={this.userListToggle}>
                <svg className="svgOnload"> 
                <symbol id="alter"><path fill="#8a8a8a" d="M100.5,3.5c-50.665,0-92.159,39.249-95.745,89h20.304C28.827,54.048,61.245,24,100.688,24
    c41.974,0,76,34.026,76,76s-34.026,76-76,76c-16.36,0-31.498-5.186-43.894-13.982c2.125-3.098,6.972-11.393,7.296-11.393
    c0.786,0,1.967-0.393,2.753-1.18l5.113-5.113l13.374-1.967c1.573-0.393,3.147-1.573,3.147-3.147l2.36-13.767l14.947-14.947
    c11.8,5.113,25.96,2.36,35.401-6.687c12.587-12.587,12.587-33.041,0-45.627c-12.587-12.587-33.041-12.587-45.627,0
    c-9.44,9.44-11.8,23.207-6.687,35.401l-38.941,38.94c-0.787,0.787-1.18,1.574-1.18,2.754s0.393,1.967,1.18,2.753L63.719,150.5
    c0.308,0.308-5.236,8.111-7.214,11.314C39.692,149.765,27.988,131.038,25.3,109.5H5.017c5.001,48.32,45.841,86,95.483,86
    c53.019,0,96-42.981,96-96S153.519,3.5,100.5,3.5z M119.551,69.99c5.507,0,9.833,4.327,9.833,9.833c0,5.507-4.327,9.833-9.833,9.833
    c-5.507,0-9.833-4.327-9.833-9.833S114.044,69.99,119.551,69.99z M61.73,136.858l-2.753-2.753l27.534-27.534l2.753,2.753
    L61.73,136.858z"/></symbol> 
                <symbol id="exit"><path fill="#ff5c68" d="M99.78,171.65c-39.805,0-71.732-32.341-71.732-71.732S60.39,28.187,99.78,28.187
    c14.927,0,28.61,4.561,40.22,12.439c14.512,0,17.415,0.415,35.244,0.415C157.829,18.65,130.463,4.553,99.78,4.553
    C46.707,4.553,4,47.26,4,100.333s42.707,95.78,95.78,95.78c30.683,0,57.634-14.098,75.463-36.488h-34.829
    C128.805,167.089,114.707,171.65,99.78,171.65z M137.463,131.431l14.512,14.512L198,99.919l-46.024-46.024l-14.512,14.512
    l19.902,19.902H73.195v23.22h84.171L137.463,131.431L137.463,131.431z"/></symbol>
            </svg>
                <div className="user-icon clearfix"><span className="glyphicon glyphicon-user" aria-hidden="true"></span></div>
                <ul className={`user-list ${active ? '' : 'hidden'}`}>
                {
                    list.map(item => <li key={item.key} onClick={()=>this.itemClick(item.key)}><svg><use xlinkHref={item.path} transform="scale(0.08,0.08)" x="0" y="0" viewBox="0 0 20 20" width="200" height="200"/></svg><span>{item.name}</span></li>)
                }
                </ul>
            </div>
        )
    }
}

UserCenter.propTypes = {
    className: PropTypes.string,
    router: PropTypes.any.isRequired
}

const mapStateToProps = (state, ownProps) => {
    return {}
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        actions: bindActionCreators({
            overlayerShow,
            overlayerHide,
            confirmExit
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(UserCenter);