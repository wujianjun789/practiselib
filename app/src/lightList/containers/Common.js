import '../../../public/styles/lightManage-list.less';
import React from 'react';
import { injectIntl } from 'react-intl';
import Content from '../../components/Content';
import SearchText from '../../components/SearchText';
import Table from '../../components/Table';
import Page from '../../components/Page';
import Select from '../../components/Select.1';
import MapView from '../../components/MapView';
import { getChildDomainList } from '../../api/domain';
import {
  getSearchAssets,
  getSearchCount,
  getDeviceStatusById
} from '../../api/asset';
import { getDeviceTypeByModel } from '../../util/index';
import Immutable from 'immutable';

export default function common(DeviceOperation) {
  class Common extends React.Component {
    state = {
      sidebarCollapse: false,
      operationCollapse: false,
      mapCollapse: false,
      domainList: {
        titleField: 'name',
        valueField: 'name',
        options: []
      },
      currentDomain: '',
      search: {
        value: '',
        placeholder: this.props.intl.formatMessage({
          id: 'lightManage.list.devicePlaceholder'
        })
      },
      page: {
        total: 0,
        current: 1,
        limit: 15
      },
      deviceList: [],
      currentDevice: '',
      selectDevice: {
        id: this.props.route.path,
        position: [],
        data: []
      }
    };
    componentWillMount() {
      this._isMounted = true;
      //根据路由参数确定模型
      this.model = this.props.route.path;
      const columns = require(`../TableData/${this.model}Table.json`);
      this.columns = columns.map(item => ({
        id: item.id,
        field: item.field,
        title: this.props.intl.formatMessage({ id: item.title })
      }));
      this.initDomain();
    }
    componentWillUnmount() {
      this._isMounted = false;
    }
    initDomain = () => {
      getChildDomainList(data => {
        if (!this._isMounted) {
          return;
        }
        let currentDomain = '';
        if (data.length) {
          currentDomain = data[0];
        }
        this.setState(
          {
            domainList: { ...this.state.domainList, options: data },
            currentDomain
          },
          this.initDevice
        );
      });
    };
    initDevice = () => {
      if (!this._isMounted) {
        return;
      }
      if (this.state.currentDomain) {
        const {
          currentDomain,
          search: { value },
          page: { current, limit }
        } = this.state;
        const offset = limit * (current - 1);
        getSearchCount(
          currentDomain.id,
          this.model,
          value.trim(),
          this._isMounted && this.updatePageSize
        );
        getSearchAssets(
          currentDomain.id,
          this.model,
          value.trim(),
          offset,
          limit,
          this._isMounted && this.updateDevice
        );
      }
    };
    updatePageSize = data => {
      this.setState({ page: { ...this.state.page, total: data.count } });
    };
    updateDevice = data => {
      const currentDevice = data.length === 0 ? '' : data[0];
      this.setState({ deviceList: data, currentDevice }, () => {
        this.updateDeviceListStatus(this.state.deviceList);
        this.updateSelectDevice(this.state.currentDevice);
      });
    };
    updateDeviceListStatus = list => {
      if (!list.length) {
        return;
      }
      let count = 0;
      let newList = [];
      list.map((item, index) => {
        getDeviceStatusById(item.id, res => {
          count++;
          for (let i of Object.keys(res.status)) {
            item[i] = res.status[i];
          }
          newList.push(item);
          if (count === newList.length) {
            this.setState({ deviceList: newList });
          }
        });
      });
    };
    //地图显示
    updateSelectDevice = item => {
      const { selectDevice } = this.state;
      selectDevice.latlng = item.geoPoint;
      selectDevice.data.splice(0);
      selectDevice.data.push({
        id: item.id,
        type: item.extend.type,
        name: item.name
      });
      selectDevice.domainId = item.domainId;
      selectDevice.position.splice(0);
      selectDevice.position.push(
        Object.assign(
          {},
          {
            device_id: item.id,
            device_type: getDeviceTypeByModel(item.extendType)
          },
          item.geoPoint
        )
      );
      this.setState({ selectDevice });
    };
    onDomainChange = e => {
      const { id, selectedIndex } = e.target;
      switch (id) {
        case 'domain':
          const currentDomain = this.state.domainList.options[selectedIndex];
          this.setState({ currentDomain }, this.initDevice);
          break;
      }
    };
    onSearchChange = v => {
      this.setState({
        search: { ...this.state.search, value: v }
      });
    };
    onSearchSubmit = () => {
      this.setState(
        {
          search: { ...this.state.search, current: 1 }
        },
        this.initDevice
      );
    };
    componentDidUpdate() {
      // console.log(this.state);
    }
    handleCollapse = id => {
      this.setState({ [id]: !this.state[id] });
    };
    onPageNumChange = page => {
      this.setState(
        { page: { ...this.state.page, current: page } },
        this.initDevice
      );
    };
    onRowClick = currentDevice => {
      this.setState({ currentDevice: currentDevice.toJS() }, () => {
        this.updateSelectDevice(this.state.currentDevice);
      });
    };
    render() {
      const {
        sidebarCollapse,
        operationCollapse,
        mapCollapse,
        domainList,
        currentDomain,
        search: { value, placeholder },
        page: { limit, current, total },
        deviceList,
        currentDevice,
        selectDevice
      } = this.state;
      const { formatMessage } = this.props.intl;
      const offset = operationCollapse ? 131 : 0;
      const top = 247 - offset;
      const disabled = deviceList.length == 0 ? true : false;
      return (
        <Content className={`list-lcc ${sidebarCollapse ? 'collapse' : ''}`}>
          <div className="content-left">
            <div className="heading">
              <Select
                id="domain"
                titleField={domainList.titleField}
                valueField={domainList.valueField}
                options={domainList.options}
                value={
                  currentDomain === ''
                    ? ''
                    : currentDomain[domainList.valueField]
                }
                onChange={this.onDomainChange}
              />
              <SearchText
                placeholder={placeholder}
                value={value}
                onChange={this.onSearchChange}
                submit={this.onSearchSubmit}
              />
            </div>
            <div className="table-container">
              <Table
                columns={this.columns}
                data={Immutable.fromJS(deviceList)}
                activeId={currentDevice.id}
                rowClick={this.onRowClick}
              />
              <Page
                className={`page ${total == 0 ? 'hidden' : ''}`}
                showSizeChanger
                pageSize={limit}
                current={current}
                total={total}
                onChange={this.onPageNumChange}
              />
            </div>
          </div>
          <div
            className={`container-fluid sidebar-info ${
              sidebarCollapse ? 'sidebar-collapse' : ''
            }`}
          >
            <div
              className="row collapse-container"
              role="presentation"
              onClick={() => this.handleCollapse('sidebarCollapse')}
            >
              <span
                className={
                  sidebarCollapse ? 'icon_horizontal' : 'icon_vertical'
                }
              />
            </div>
            <div className="panel panel-default panel-custom">
              <div
                className="panel-heading"
                role="presentation"
                onClick={() => this.handleCollapse('operationCollapse')}
              >
                <span className="icon_touch" />
                {formatMessage({
                  id: 'lightManage.list.operation'
                })}
                <span
                  class={`icon pull-right glyphicon ${
                    operationCollapse
                      ? 'glyphicon-chevron-down'
                      : 'glyphicon-chevron-up'
                  }`}
                />
              </div>
              <div
                class={`panel-body device-operation-panel ${
                  operationCollapse ? 'device-hidden' : ''
                }`}
              >
                <DeviceOperation
                  disabled={disabled}
                  currentDevice={currentDevice}
                />
              </div>
            </div>
            <div class="panel panel-default">
              <div
                class="panel-heading"
                role="presentation"
                onClick={() => this.handleCollapse('mapCollapse')}
              >
                <span class="icon_map" />
                <span>
                  {formatMessage({
                    id: 'lightManage.list.mapPosition'
                  })}
                </span>
                <span
                  class={`icon pull-right glyphicon ${
                    mapCollapse
                      ? 'glyphicon-chevron-down'
                      : 'glyphicon-chevron-up'
                  }`}
                />
              </div>
              <div
                class={`panel-body map-container ${
                  mapCollapse ? 'device-hidden' : ''
                }`}
                style={{ top: top }}
              >
                <MapView option={{ mapZoom: false }} mapData={selectDevice} />
              </div>
            </div>
          </div>
        </Content>
      );
    }
  }
  return injectIntl(Common);
}
