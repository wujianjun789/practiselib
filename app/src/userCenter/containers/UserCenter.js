/**
 * create by Azrael on 2017/7/4
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
/**
 * @param {String} className  optional
 */
export class UserCenter extends Component{
    constructor(props) {
        super(props);
        this.state = {
            active: false,
            list: [
                {name: '修改密码', key: 'alter'},
                {name: '退出管理系统', key: 'exit'}
            ]
        }
        this.userListToggle = this.userListToggle.bind(this);
        this.itemClick = this.itemClick.bind(this);
    }

    userListToggle() {
        this.setState((prevState)=> {
            return {active: !prevState.active};
        });
    }

    itemClick(key) {

    }

    render() {
        const {active, list} = this.state;
        const {className=""} = this.props;
        return (
            <div className={`user-center ${className}`} onClick={this.userListToggle}>
                <div className="user-icon clearfix"><span className="icon icon-usr"></span></div>
                <ul className={`user-list ${active ? '' : 'hidden'}`}>
                {
                    list.map(item => <li key={item.key} onClick={(key)=>this.itemClick(key)}><span className={`icon icon-${item.key}`}></span>{item.name}</li>)
                }
                </ul>
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        actions: bindActionCreators({

        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(UserCenter);

UserCenter.propTypes = {
    className: PropTypes.string
}