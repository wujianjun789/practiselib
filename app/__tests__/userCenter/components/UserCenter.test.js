import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {UserCenter} from '../../../src/common/containers/UserCenter';

describe('<UserCenter />', () => {
    const router = {push: jest.fn()};
    it('render with props.className=undefined', () => {
        const userCenter = shallow(<UserCenter router={router} />);
        const container = userCenter.find('div.user-center');
        expect(container.length).toBe(1);

        const userIcon = container.find('.user-icon.clearfix');
        expect(userIcon.length).toBe(1);

        const icon = userIcon.find('.glyphicon.glyphicon-user');
        expect(icon.length).toBe(1);

        const _ul = container.find('ul.user-list');
        expect(_ul.length).toBe(1);

        const _li = _ul.find('li');
        expect(_li.length).toBe(2);

        expect(_li.at(0).text()).toBe('修改密码');
        expect(_li.at(1).text()).toBe('退出管理系统');
    });

    it('render with props.className="user_center"', () => {
        const userCenter = shallow(<UserCenter className="user_center" router={router} />);
        const container = userCenter.find('div.user-center');
        expect(container.length).toBe(1);
        expect(container.prop('className')).toBe('user-center user_center');
    });

    it('snapshot', () => {
        const userCenter = renderer.create(<UserCenter router={router}/>);
        let tree = userCenter.toJSON();

        expect(tree).toMatchSnapshot();
    });
});