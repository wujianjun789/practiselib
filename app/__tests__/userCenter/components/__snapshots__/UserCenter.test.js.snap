exports[`<UserCenter /> user Info list toggle snapshot 1`] = `
<div
  className="user-center "
  onClick={[Function]}>
  <svg
    className="svgOnload">
    <symbol
      id="alter">
      <path
        d="M100.5,3.5c-50.665,0-92.159,39.249-95.745,89h20.304C28.827,54.048,61.245,24,100.688,24 c41.974,0,76,34.026,76,76s-34.026,76-76,76c-16.36,0-31.498-5.186-43.894-13.982c2.125-3.098,6.972-11.393,7.296-11.393 c0.786,0,1.967-0.393,2.753-1.18l5.113-5.113l13.374-1.967c1.573-0.393,3.147-1.573,3.147-3.147l2.36-13.767l14.947-14.947 c11.8,5.113,25.96,2.36,35.401-6.687c12.587-12.587,12.587-33.041,0-45.627c-12.587-12.587-33.041-12.587-45.627,0 c-9.44,9.44-11.8,23.207-6.687,35.401l-38.941,38.94c-0.787,0.787-1.18,1.574-1.18,2.754s0.393,1.967,1.18,2.753L63.719,150.5 c0.308,0.308-5.236,8.111-7.214,11.314C39.692,149.765,27.988,131.038,25.3,109.5H5.017c5.001,48.32,45.841,86,95.483,86 c53.019,0,96-42.981,96-96S153.519,3.5,100.5,3.5z M119.551,69.99c5.507,0,9.833,4.327,9.833,9.833c0,5.507-4.327,9.833-9.833,9.833 c-5.507,0-9.833-4.327-9.833-9.833S114.044,69.99,119.551,69.99z M61.73,136.858l-2.753-2.753l27.534-27.534l2.753,2.753 L61.73,136.858z"
        fill="#8a8a8a" />
    </symbol>
    <symbol
      id="exit">
      <path
        d="M99.78,171.65c-39.805,0-71.732-32.341-71.732-71.732S60.39,28.187,99.78,28.187 c14.927,0,28.61,4.561,40.22,12.439c14.512,0,17.415,0.415,35.244,0.415C157.829,18.65,130.463,4.553,99.78,4.553 C46.707,4.553,4,47.26,4,100.333s42.707,95.78,95.78,95.78c30.683,0,57.634-14.098,75.463-36.488h-34.829 C128.805,167.089,114.707,171.65,99.78,171.65z M137.463,131.431l14.512,14.512L198,99.919l-46.024-46.024l-14.512,14.512 l19.902,19.902H73.195v23.22h84.171L137.463,131.431L137.463,131.431z"
        fill="#ff5c68" />
    </symbol>
  </svg>
  <div
    className="user-icon clearfix">
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-user" />
  </div>
  <ul
    className="user-list hidden">
    <li
      onClick={[Function]}>
      <svg
        version="1.0"
        xmlns="http://www.w3.org/2000/svg">
        <use
          height="200"
          transform="scale(0.08,0.08)"
          viewBox="0 0 20 20"
          width="200"
          x="0"
          xlinkHref="#alter"
          y="0" />
      </svg>
      <span>
        修改密码
      </span>
    </li>
    <li
      onClick={[Function]}>
      <svg
        version="1.0"
        xmlns="http://www.w3.org/2000/svg">
        <use
          height="200"
          transform="scale(0.08,0.08)"
          viewBox="0 0 20 20"
          width="200"
          x="0"
          xlinkHref="#exit"
          y="0" />
      </svg>
      <span>
        退出管理系统
      </span>
    </li>
  </ul>
</div>
`;

exports[`<UserCenter /> user Info list toggle snapshot 2`] = `
<div
  className="user-center "
  onClick={[Function]}>
  <svg
    className="svgOnload">
    <symbol
      id="alter">
      <path
        d="M100.5,3.5c-50.665,0-92.159,39.249-95.745,89h20.304C28.827,54.048,61.245,24,100.688,24 c41.974,0,76,34.026,76,76s-34.026,76-76,76c-16.36,0-31.498-5.186-43.894-13.982c2.125-3.098,6.972-11.393,7.296-11.393 c0.786,0,1.967-0.393,2.753-1.18l5.113-5.113l13.374-1.967c1.573-0.393,3.147-1.573,3.147-3.147l2.36-13.767l14.947-14.947 c11.8,5.113,25.96,2.36,35.401-6.687c12.587-12.587,12.587-33.041,0-45.627c-12.587-12.587-33.041-12.587-45.627,0 c-9.44,9.44-11.8,23.207-6.687,35.401l-38.941,38.94c-0.787,0.787-1.18,1.574-1.18,2.754s0.393,1.967,1.18,2.753L63.719,150.5 c0.308,0.308-5.236,8.111-7.214,11.314C39.692,149.765,27.988,131.038,25.3,109.5H5.017c5.001,48.32,45.841,86,95.483,86 c53.019,0,96-42.981,96-96S153.519,3.5,100.5,3.5z M119.551,69.99c5.507,0,9.833,4.327,9.833,9.833c0,5.507-4.327,9.833-9.833,9.833 c-5.507,0-9.833-4.327-9.833-9.833S114.044,69.99,119.551,69.99z M61.73,136.858l-2.753-2.753l27.534-27.534l2.753,2.753 L61.73,136.858z"
        fill="#8a8a8a" />
    </symbol>
    <symbol
      id="exit">
      <path
        d="M99.78,171.65c-39.805,0-71.732-32.341-71.732-71.732S60.39,28.187,99.78,28.187 c14.927,0,28.61,4.561,40.22,12.439c14.512,0,17.415,0.415,35.244,0.415C157.829,18.65,130.463,4.553,99.78,4.553 C46.707,4.553,4,47.26,4,100.333s42.707,95.78,95.78,95.78c30.683,0,57.634-14.098,75.463-36.488h-34.829 C128.805,167.089,114.707,171.65,99.78,171.65z M137.463,131.431l14.512,14.512L198,99.919l-46.024-46.024l-14.512,14.512 l19.902,19.902H73.195v23.22h84.171L137.463,131.431L137.463,131.431z"
        fill="#ff5c68" />
    </symbol>
  </svg>
  <div
    className="user-icon clearfix">
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-user" />
  </div>
  <ul
    className="user-list ">
    <li
      onClick={[Function]}>
      <svg
        version="1.0"
        xmlns="http://www.w3.org/2000/svg">
        <use
          height="200"
          transform="scale(0.08,0.08)"
          viewBox="0 0 20 20"
          width="200"
          x="0"
          xlinkHref="#alter"
          y="0" />
      </svg>
      <span>
        修改密码
      </span>
    </li>
    <li
      onClick={[Function]}>
      <svg
        version="1.0"
        xmlns="http://www.w3.org/2000/svg">
        <use
          height="200"
          transform="scale(0.08,0.08)"
          viewBox="0 0 20 20"
          width="200"
          x="0"
          xlinkHref="#exit"
          y="0" />
      </svg>
      <span>
        退出管理系统
      </span>
    </li>
  </ul>
</div>
`;

exports[`<UserCenter /> user Info list toggle snapshot 3`] = `
<div
  className="user-center "
  onClick={[Function]}>
  <svg
    className="svgOnload">
    <symbol
      id="alter">
      <path
        d="M100.5,3.5c-50.665,0-92.159,39.249-95.745,89h20.304C28.827,54.048,61.245,24,100.688,24 c41.974,0,76,34.026,76,76s-34.026,76-76,76c-16.36,0-31.498-5.186-43.894-13.982c2.125-3.098,6.972-11.393,7.296-11.393 c0.786,0,1.967-0.393,2.753-1.18l5.113-5.113l13.374-1.967c1.573-0.393,3.147-1.573,3.147-3.147l2.36-13.767l14.947-14.947 c11.8,5.113,25.96,2.36,35.401-6.687c12.587-12.587,12.587-33.041,0-45.627c-12.587-12.587-33.041-12.587-45.627,0 c-9.44,9.44-11.8,23.207-6.687,35.401l-38.941,38.94c-0.787,0.787-1.18,1.574-1.18,2.754s0.393,1.967,1.18,2.753L63.719,150.5 c0.308,0.308-5.236,8.111-7.214,11.314C39.692,149.765,27.988,131.038,25.3,109.5H5.017c5.001,48.32,45.841,86,95.483,86 c53.019,0,96-42.981,96-96S153.519,3.5,100.5,3.5z M119.551,69.99c5.507,0,9.833,4.327,9.833,9.833c0,5.507-4.327,9.833-9.833,9.833 c-5.507,0-9.833-4.327-9.833-9.833S114.044,69.99,119.551,69.99z M61.73,136.858l-2.753-2.753l27.534-27.534l2.753,2.753 L61.73,136.858z"
        fill="#8a8a8a" />
    </symbol>
    <symbol
      id="exit">
      <path
        d="M99.78,171.65c-39.805,0-71.732-32.341-71.732-71.732S60.39,28.187,99.78,28.187 c14.927,0,28.61,4.561,40.22,12.439c14.512,0,17.415,0.415,35.244,0.415C157.829,18.65,130.463,4.553,99.78,4.553 C46.707,4.553,4,47.26,4,100.333s42.707,95.78,95.78,95.78c30.683,0,57.634-14.098,75.463-36.488h-34.829 C128.805,167.089,114.707,171.65,99.78,171.65z M137.463,131.431l14.512,14.512L198,99.919l-46.024-46.024l-14.512,14.512 l19.902,19.902H73.195v23.22h84.171L137.463,131.431L137.463,131.431z"
        fill="#ff5c68" />
    </symbol>
  </svg>
  <div
    className="user-icon clearfix">
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-user" />
  </div>
  <ul
    className="user-list hidden">
    <li
      onClick={[Function]}>
      <svg
        version="1.0"
        xmlns="http://www.w3.org/2000/svg">
        <use
          height="200"
          transform="scale(0.08,0.08)"
          viewBox="0 0 20 20"
          width="200"
          x="0"
          xlinkHref="#alter"
          y="0" />
      </svg>
      <span>
        修改密码
      </span>
    </li>
    <li
      onClick={[Function]}>
      <svg
        version="1.0"
        xmlns="http://www.w3.org/2000/svg">
        <use
          height="200"
          transform="scale(0.08,0.08)"
          viewBox="0 0 20 20"
          width="200"
          x="0"
          xlinkHref="#exit"
          y="0" />
      </svg>
      <span>
        退出管理系统
      </span>
    </li>
  </ul>
</div>
`;
